Use better defaults in the clang-tidy wrapper script

1ntEgr8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61674
* #61711
* #61689
* #61672
* **#61651**


This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)
- `generate_build_files` can now be run as a script

Differential Revision: [D29743661](https://our.internmc.facebook.com/intern/diff/D29743661)

1ntEgr8 committed:
Use better defaults in the clang-tidy wrapper script

[ghstack-poisoned]

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)


[ghstack-poisoned]

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)


[ghstack-poisoned]

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)


[ghstack-poisoned]

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)


[ghstack-poisoned]

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)


[ghstack-poisoned]

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)


[ghstack-poisoned]

driazati commented on tools/linter/clang_tidy/__main__.py:
```diff
@@ -173,17 +180,16 @@ def main() -> None:
         )
         print(msg)
-        exit(1)
+        raise RuntimeError("Could not find clang-tidy binary")
```
nit: why raise 2 errors?
```suggestion
        raise RuntimeError(msg)
```

driazati commented on .github/workflows/lint.yml:
```diff
@@ -335,12 +335,11 @@ jobs:
-            --verbose \
-            "$@" >"${GITHUB_WORKSPACE}"/clang-tidy-output.txt
-          cat "${GITHUB_WORKSPACE}"/clang-tidy-output.txt
+            --verbose 2>&1 | tee "${GITHUB_WORKSPACE}"/clang-tidy-output.txt
```
```suggestion
            2>&1 | tee "${GITHUB_WORKSPACE}"/clang-tidy-output.txt
```

do we need `--verbose` in CI? it's nice for us but not very useful for a random developer looking for their failures https://github.com/pytorch/pytorch/pull/61674/checks?check_run_id=3072761257

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)
- `generate_build_files` can now be run as a script


[ghstack-poisoned]

janeyx99 commented on tools/linter/clang_tidy/__main__.py:
```diff
@@ -168,26 +175,27 @@ def parse_args() -> argparse.Namespace:
             "You can install it by running:\n"
             "   python3 tools/linter/install/clang_tidy.py"
         )
         print(msg)
```
```suggestion
```
Following David's suggestion below

1ntEgr8 committed:
Update on "Use better defaults in the clang-tidy wrapper script"



This PR sets some QOL defaults to the clang-tidy wrapper script and refactors how defaults are set.

- Runs in parallel
- Custom executable (prints an error message to users asking them to install our custom build)
- `generate_build_files` can now be run as a script


[ghstack-poisoned]

1ntEgr8 commented:
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29743661).
